### PR TITLE
refactor: rename submariner to submariner-gateway

### PR DESCRIPTION
**Pull request by SteveMattar**
Use a more understandable name to distinguish between the project and the component.

:warning: Needs to push `quay.io/submariner/submariner-gateway:devel` first :warning:

BREAKING CHANGE

Signed-off-by: Steve Mattar <smattar@redhat.com>

### CODE REVIEW

**Review by dfarrell07 — DISMISSED** [review]
This is a great bit of refactoring to finally handle, thanks for working on this @SteveMattar!

I have some **very** old FIXMEs in the original operator verifications that should now be removed. :partying_face: 

```
# FIXME A better name might be submariner-engine, but just kinda-matching submariner-<random hash> name used by Helm/upstream tests
```

https://github.com/submariner-io/submariner-operator/blob/master/scripts/kind-e2e/lib_operator_verify_subm.sh#L19

**Comment by mangelajo** [comment]
I would propose renaming to submariner-gateway instead.

**Comment by SteveMattar** [comment]
@mangelajo would you kindly tag and push `quay.io/submariner/submariner-gateway:devel` in favor of our tests?

**Comment by skitt** [comment]
Instead of manually pushing images, we need the following changes:

* make Submariner build the images as `submariner` and `submariner-gateway`
* make Shipyard pull both images for e2e
* this PR should then pass
* once this PR is merged, we can drop the first two changes

**Comment by mangelajo** [comment]
> Instead of manually pushing images, we need the following changes:
> 
> * make Submariner build the images as `submariner` and `submariner-gateway`
> * make Shipyard pull both images for e2e
> * this PR should then pass
> * once this PR is merged, we can drop the first two changes

1st step: https://github.com/submariner-io/submariner/pull/1154

File: scripts/kind-e2e/lib_operator_verify_subm.sh [hunk 1]
```diff
@@ -379,18 +377,18 @@ function verify_subm_engine_container() {
...
-    # Verify the engine entry script is in the expected place and in PATH
-    kubectl exec $subm_engine_pod_name --namespace=$subm_ns -- command -v submariner.sh | grep /usr/local/bin/submariner.sh
-  elif kubectl exec $subm_engine_pod_name --namespace=$subm_ns -- which which; then
-    # Verify the engine binary is in the expected place and in PATH
-    kubectl exec $subm_engine_pod_name --namespace=$subm_ns -- which submariner-engine | grep /usr/local/bin/submariner-engine
+    # Verify the gateway entry script is in the expected place and in PATH
+    kubectl exec $subm_gateway_pod_name --namespace=$subm_ns -- command -v submariner.sh | grep /usr/local/bin/submariner.sh
+  elif kubectl exec $subm_gateway_pod_name --namespace=$subm_ns -- which which; then
+    # Verify the gateway binary is in the expected place and in PATH
+    kubectl exec $subm_gateway_pod_name --namespace=$subm_ns -- which submariner-gateway | grep /usr/local/bin/submariner-gateway
```
Comment (mangelajo) [inline scripts/kind-e2e/lib_operator_verify_subm.sh]: This line shouldn't change

As the binary inside the image remains as submariner-engine yet.

```suggestion
    kubectl exec $subm_gateway_pod_name --namespace=$subm_ns -- which submariner-engine | grep /usr/local/bin/submariner-engine
```

Comment (mangelajo) [inline scripts/kind-e2e/lib_operator_verify_subm.sh]: We could probably remove the line altogether, I believe checking may be for submariner.sh is more than enough.

File: scripts/kind-e2e/lib_operator_verify_subm.sh [hunk 2]
```diff
@@ -379,18 +377,18 @@ function verify_subm_engine_container() {
   grep "SUBMARINER_NATENABLED=$natEnabled" $env_file
   grep "HOME=/root" $env_file
 
-  if kubectl exec $subm_engine_pod_name --namespace=$subm_ns -- command -v command; then
-    # Verify the engine binary is in the expected place and in PATH
-    kubectl exec $subm_engine_pod_name --namespace=$subm_ns -- command -v submariner-engine | grep /usr/local/bin/submariner-engine
+  if kubectl exec $subm_gateway_pod_name --namespace=$subm_ns -- command -v command; then
+    # Verify the gateway binary is in the expected place and in PATH
+    kubectl exec $subm_gateway_pod_name --namespace=$subm_ns -- command -v submariner-gateway | grep /usr/local/bin/submariner-gateway
```
Comment (mangelajo) [inline scripts/kind-e2e/lib_operator_verify_subm.sh]:
```suggestion
    kubectl exec $subm_gateway_pod_name --namespace=$subm_ns -- command -v submariner-engine | grep /usr/local/bin/submariner-engine
```

File: scripts/kind-e2e/lib_operator_verify_subm.sh (hunk 1 above)
Comment (SteveMattar) [reply scripts/kind-e2e/lib_operator_verify_subm.sh]: Right, we can rename that later after we merge https://github.com/submariner-io/submariner/pull/1160 & https://github.com/submariner-io/shipyard/pull/462

Comment (SteveMattar) [reply scripts/kind-e2e/lib_operator_verify_subm.sh]: Right, we can rename that later after we merge submariner-io/submariner#1160 & submariner-io/shipyard#462

**Comment by mangelajo** [comment]
Ok, upgrade test failed and we will need to update that in shipyard I guess:

```
Error response from daemon: No such image: quay.io/submariner/submariner-gateway:latest
[submariner-operator]$ docker pull quay.io/submariner/submariner-gateway:latest
Error response from daemon: manifest for quay.io/submariner/submariner-gateway:latest not found: manifest unknown: manifest unknown
make[1]: *** [Makefile:154: preload-images] Error 1
make[1]: Leaving directory '/go/src/github.com/submariner-io/submariner-operator'

```
But other than that it looks good

**Review by mangelajo — APPROVED** [review]

**Review by nyechiel — APPROVED** [review]

**Comment by nyechiel** [comment]
@dfarrell07 can you assess any docs impact for this change? I don't think there is any, but I might be wrong.

**Comment by nyechiel** [comment]
> @dfarrell07 can you assess any docs impact for this change? I don't think there is any, but I might be wrong.

Ping @dfarrell07 :)

**Pull request merged**